.github/workflows/e2e_custom_cl.yml

```diff
@@ -19,13 +19,35 @@ jobs:
         RUSTUP_HOME: "/root/.rustup"
         FORCE_COLOR: 1
     steps:
+      - name: Install latest Git version (>= 2.18.0) for actions/checkout
+        run: |
+          apt-get install software-properties-common -y
+          add-apt-repository ppa:git-core/ppa
+          apt update
+          apt install git -y
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
       - uses: actions/checkout@f25a3a9f25bd5f4c5d77189cab02ff357b5aedeb # v2.4.1
+      - uses: smartcontractkit/tool-versions-to-env-action@v1.0.7
+        id: tool-versions
+      - name: Setup go ${{ steps.tool-versions.outputs.golang_version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.tool-versions.outputs.golang_version }}
       - run: echo "$HOME"
       - run: echo "$PATH"
       - run: yarn install --frozen-lockfile
+      - name: Generate build artifacts for go bindings check
+        run: anchor build
+        working-directory: contracts
+      - name: Check generated go bindings are up to date
+        run: |
+          go install github.com/gagliardetto/anchor-go@v0.2.3
+          ./scripts/anchor-go-gen.sh
+          git diff --stat --exit-code
```

Comment:
added to the e2e artifact build because if the go bindings are not up to date, the e2e tests are the main impact